scalafix-sbt/src/sbt-test/sbt-scalafix/cross-build/test

```diff
@@ -1,15 +1,33 @@
 # compile should not affect scalafix
 > compile
--> scalafixTest ProcedureSyntax
-> scalafix ProcedureSyntax
-> scalafixTest ProcedureSyntax
-> javaProject/scalafix ProcedureSyntax
+
+-> compile:scalafixTest ProcedureSyntax
+-> test:scalafixTest ProcedureSyntax
+-> it:scalafixTest ProcedureSyntax
+
+> compile:scalafix ProcedureSyntax
+> test:scalafix ProcedureSyntax
+> it:scalafix ProcedureSyntax
+
+> compile:scalafixTest ProcedureSyntax
+> test:scalafixTest ProcedureSyntax
+> it:scalafixTest ProcedureSyntax
```

Comment:
This does not test that `compile:scalafix` is independent from `test:scalafix`

What about:
```
-> compile:scalafixTest ProcedureSyntax
-> test:scalafixTest ProcedureSyntax
-> it:scalafixTest ProcedureSyntax

> compile:scalafix ProcedureSyntax
> compile:scalafixTest ProcedureSyntax
-> test:scalafixTest ProcedureSyntax
-> it:scalafixTest ProcedureSyntax

> test:scalafix ProcedureSyntax
> test:scalafixTest ProcedureSyntax
-> it:scalafixTest ProcedureSyntax

> it:scalafix ProcedureSyntax
> it:scalafixTest ProcedureSyntax
```